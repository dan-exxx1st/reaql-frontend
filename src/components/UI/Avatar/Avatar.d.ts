declare namespace NAvatar {
    interface IProps {
        src?: HTMLImageElement['src'];
        alt?: string;
    }
}
