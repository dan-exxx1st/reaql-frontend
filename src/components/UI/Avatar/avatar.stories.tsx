import React from 'react';
import Avatar from '.';

export default {
	title: 'UI/Avatar',
};

export const AvatarWithDefaultImage = () => <Avatar alt="Ivan Markov" />;

export const AvatarWithUrl = () => (
	<Avatar
		src="https://www.thespruce.com/thmb/wgUT9rFyV6H6dREu2ErfcacebgU=/2048x1722/filters:fill(auto,1)/House_Sparrow_England_-_May_09-5c5d2da4c9e77c0001662023.jpg"
		alt="Ivan Markov"
	/>
);
