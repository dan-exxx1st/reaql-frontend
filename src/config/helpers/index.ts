import { useState, useEffect } from 'react';

export function useLoaded({ src }: { src?: HTMLImageElement['src'] }) {
	const [loaded, setLoaded] = useState('false');

	useEffect(() => {
		if (!src) {
			return undefined;
		}

		setLoaded('loading');

		let active = true;
		const image = new Image();
		image.src = src;
		image.onload = () => {
			if (!active) {
				return;
			}
			setLoaded('loaded');
		};
		image.onerror = () => {
			if (!active) {
				return;
			}
			setLoaded('error');
		};

		return () => {
			active = false;
		};
	}, [src]);

	return loaded;
}
