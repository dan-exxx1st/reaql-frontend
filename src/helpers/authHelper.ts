import { Session, User } from 'lib/graphql/types';
import { useContext } from 'react';
import { UserContext } from './contexts/userContext';

export const CheckAuth = () => {
    const { state } = useContext(UserContext);
    const sessionString = localStorage.getItem('session');
    if (sessionString || state?.user?.email) {
        return true;
    }
    return false;
};

export const getAuthTokens = () => {
    const session = localStorage.getItem('session');
    if (session) {
        const {
            accessToken,
            refreshToken,
        }: { accessToken: string; refreshToken: string } = JSON.parse(session);

        return { accessToken, refreshToken };
    } else {
        return { accessToken: '', refreshToken: '' };
    }
};

export const setSession = (session: {
    refreshToken: string;
    accessToken: string;
}) => {
    const sessionData = {
        accessToken: session.accessToken,
        refreshToken: session.refreshToken,
    };
    const sessionString = JSON.stringify(sessionData);
    localStorage.setItem('session', sessionString);
};

export const setUserAndSession = (payload: {
    user: User;
    session?: Session | null;
}) => {
    const { user, session } = payload;
    if (session) setSession(session);
    localStorage.setItem('userEmail', user.email);
};

export const clearSession = () => {
    localStorage.clear();
    window.location.reload();
};
