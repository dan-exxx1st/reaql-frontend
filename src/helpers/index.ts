import { parse } from 'querystring';

export function getDialogIdFromSearch(search: string) {
    const params = parse(search);

    const dialogId = Object.entries(params)
        .map(([key, value]) => key.indexOf('dialog') !== -1 && value)
        .toString();

    return dialogId;
}
