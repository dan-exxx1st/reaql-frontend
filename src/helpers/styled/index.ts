import colorTheme from './colors';
import fontTheme from './fonts';

export { colorTheme, fontTheme };

const effects = {
    formBoxShadow: '5px 5px 15px rgba(37, 37, 37, 0.1)',
};

export default {
    ...colorTheme,
    ...fontTheme,
    ...effects,
};
