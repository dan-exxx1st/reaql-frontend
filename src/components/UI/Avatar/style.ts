import styled from 'styled-components';

export const StyledImage = styled.img`
	width: 40px;
	height: 40px;
	border-radius: 50%;
`;

export const StyledAvatar = (Avatar: React.FC<NAvatar.IProps>) => styled(
	Avatar,
)<NAvatar.IProps>`
	display: flex;

	width: 40px;
	height: 40px;
`;
