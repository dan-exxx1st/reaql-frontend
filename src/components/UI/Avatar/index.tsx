import React, { FC } from 'react';
import { useLoaded } from 'config/helpers';
import { Icon, SkeletonLoader } from '..';
import { StyledImage, StyledAvatar } from './style';

const Avatar: FC<NAvatar.IProps> = ({ src, alt, className = '' }) => {
    let children = null;
    const loaded = useLoaded({ src });
    const hasImgNotFailing = src && loaded !== 'error';
    if (hasImgNotFailing) {
        children =
            loaded === 'loaded' ? (
                <StyledImage
                    className={className}
                    src={src}
                    alt={alt}
                    loading="lazy"
                />
            ) : (
                <SkeletonLoader className={className} variant="avatar" />
            );
    } else {
        children = <Icon className={className} iconName="person" />;
    }
    return <>{children}</>;
};

export default StyledAvatar(Avatar);
