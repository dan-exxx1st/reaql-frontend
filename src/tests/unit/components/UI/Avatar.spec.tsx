import React from 'react';
import { mountWithTheme, shallowWithTheme } from 'tests/helpers/withTheme';

import Avatar from 'components/UI/Avatar';

jest.mock('config/helpers', () => ({
    useLoaded: jest.fn().mockReturnValue('loaded'),
}));

describe('<Avatar />', () => {
    describe('Snapshots', () => {
        it('Avatar without a image to match snapshot', () => {
            const wrapper = mountWithTheme(
                <Avatar alt="Avatar without image" />
            );
            expect(wrapper).toMatchSnapshot();
        });

        it('Avatar with a image to match snapshot', () => {
            const wrapper = shallowWithTheme(
                <Avatar
                    src="https://yandex.ru/images/search?text=foto&pos=3&img_url=https%3A%2F%2Fget.wallhere.com%2Fphoto%2F2700x1800-px-building-castle-forest-Hohenzollern-landscape-1077207.jpg&rpt=simage"
                    alt="Avatar without image"
                />
            );
            expect(wrapper).toMatchSnapshot();
        });
    });
});
