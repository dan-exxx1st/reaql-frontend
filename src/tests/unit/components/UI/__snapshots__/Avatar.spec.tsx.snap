// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> Snapshots Avatar with a image to match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "dc": "#252525",
      "dgc": "#A5A5A5",
      "dsc": "#555555",
      "ff": "Roboto",
      "fontSizes": Object {
        "body1": "16px",
        "body2": "14px;",
        "caption1": "12px",
        "caption2": "12px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
      },
      "fontWeights": Object {
        "body1": "normal",
        "body2": "normal",
        "caption1": "normal",
        "caption2": "bold",
        "h1": "300",
        "h2": "300",
        "h3": "normal",
        "h4": "normal",
        "h5": "normal",
        "h6": "500",
      },
      "gc": "#E5E5E5",
      "letterSpacing": Object {
        "body1": "0.5px",
        "body2": "0.25px",
        "caption1": "0.4px",
        "caption2": "0.4px",
        "h1": "-1.5px",
        "h2": "-0.5px",
        "h3": "0.25px",
        "h6": "0.15px",
      },
      "lineHeight": Object {
        "body1": "19px",
        "caption1": "14px",
        "caption2": "14px",
        "h1": "112px",
        "h2": "70px",
        "h3": "56px",
        "h4": "40px",
        "h5": "28px",
        "h6": "23px",
      },
      "pc": "#00ADFF",
      "plc": "#A4E2FF",
      "sc": "#E6E6E6",
      "wc": "#FAFAFA",
    }
  }
>
  <Styled(Avatar)
    alt="Avatar without image"
    src="https://yandex.ru/images/search?text=foto&pos=3&img_url=https%3A%2F%2Fget.wallhere.com%2Fphoto%2F2700x1800-px-building-castle-forest-Hohenzollern-landscape-1077207.jpg&rpt=simage"
  />
</ThemeProvider>
`;

exports[`<Avatar /> Snapshots Avatar without a image to match snapshot 1`] = `
.c0 {
  width: 40px;
  height: 40px;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0;
  margin: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 40px;
  height: 40px;
}

<img
  alt="Icon component"
  className="c0 c1"
  src="Person.svg"
/>
`;
